### PR TITLE
add tailwind and remove bootstrap and sass

### PR DESCRIPTION
**Pull request by oxyc**
This PR removes bootstrap and sass as well as adds Tailwind and integrates `LaravelTailwindConfigServiceProvider` that provides a `tailwind()` utility function for accessing variables from `tailwind.js`.

This is just the basic support without any default styling, examples.

This isn't 100% complete:
- [ ] wait for approvedio/laravel-tailwind-config v0.1.1 to reach packagist with laravel 5.8 support
- [ ] remove composer.lock as it's too difficult to merge with upstream
- [ ] commit packages-lock.json and remove yarn.lock
- [ ] remove sass dependency from packages.json
- [ ] Revamp tailwind variables as a larger stage.
- [ ] Add templates using the tailwind classes
- [ ] Add gutenberg components and utilities.

Next up I'll add Gutenberg so where examples of usage will be seen.

### CODE REVIEW

File: app/Providers/LaravelTailwindConfigServiceProvider.php [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+{
+    /**
+     * {@inheritDoc}
+     */
+    public function register()
+    {
+        $this->registerTailwindService();
+
+        // @todo if Acorn registers globals before register providers we can
+        // remove this overload
```
Comment (oxyc) [inline app/Providers/LaravelTailwindConfigServiceProvider.php:17]: At the moment https://github.com/roots/acorn isn't entirely compatible with Laravel packages as global functions such as `base_path()` aren't available until after providers have loaded. This requires us to swap out the provider for our own and remove the code that uses these functions. We could also have used `Roots\` namespaced functions here but the functionality lost isn't so important. I brought it up on Roots slack channel and they _might_ be working on a fix, in which case we can remove this provider and use the upstream one directly.

**Comment by oxyc** [comment]
Noteworthy is Sages own PR for tailwind as the default https://github.com/roots/sage/pull/2197

This keeps using SASS tough. What do you think? Should we keep sass like they do?

**Comment by silentnoodlemaster** [comment]
> Noteworthy is Sages own PR for tailwind as the default [roots#2197](https://github.com/roots/sage/pull/2197)
> 
> This keeps using SASS tough. What do you think? Should we keep sass like they do?

I feel like a handful of small postCSS plugins is a more elegant solution. I would maybe stick to SASS if staying similar to upstream was a high priority, not for the features it brings :thinking:

**Comment by oxyc** [comment]
> > Noteworthy is Sages own PR for tailwind as the default [roots#2197](https://github.com/roots/sage/pull/2197)
> > This keeps using SASS tough. What do you think? Should we keep sass like they do?
> 
> I feel like a handful of small postCSS plugins is a more elegant solution. I would maybe stick to SASS if staying similar to upstream was a high priority, not for the features it brings 🤔

Exactly, for me it would mostly be to stick to what upstream does as much as possible. Albeit this is a very minor change so i'm not very worried about it.

**Comment by kimlehtinen** [comment]
> Noteworthy is Sages own PR for tailwind as the default [roots#2197](https://github.com/roots/sage/pull/2197)
> 
> This keeps using SASS tough. What do you think? Should we keep sass like they do?

I don't prefer one more than the other, so I'm okay with either :)

File: config/app.php [hunk 2]
```diff
@@ -69,6 +69,8 @@
 
     'providers' => [
         App\Providers\ThemeServiceProvider::class,
+        // ApprovedDigital\LaravelTailwindConfig\LaravelTailwindConfigServiceProvider::class,
```
Comment (toffebjorkskog) [inline config/app.php:72]: What is this

Comment (toffebjorkskog) [reply config/app.php:72]: nvm, i read the previous comment again,

File: package.json [hunk 3]
```diff
@@ -22,17 +22,23 @@
...
+    "postcss-preset-env": "^6.6.0",
+    "postcss-scss": "^2.0.0",
     "rimraf": "^2.6",
     "sass": "^1.19.0",
     "sass-loader": "7.*",
     "stylelint": "^9.9",
     "stylelint-config-standard": "^18.2",
-    "vue-template-compiler": "^2.6.10"
+    "tailwindcss": "^1.0.1",
+    "vue-template-compiler": "^2.6.10",
```
Comment (toffebjorkskog) [inline package.json:38]: Will we want to support vue & react & jQuery.
What is the best way to be able to pick which one you want.

1. Include all, and only import the ones you need
2. Comment away the ones you don't need
3. Only add the libraries buy need when you need them, but with documentation on how to add them

**Review by toffebjorkskog — APPROVED** [review]

File: package.json (hunk 3 above)
Comment (silentnoodlemaster) [reply package.json:38]: perhaps option 3

Comment (oxyc) [reply package.json:38]: This is actually from the sage branch https://github.com/roots/sage/commit/4f9222bae5194ccfc420d371dd30f77f025d8ec8

We could remove it if we wanted to though. It's only for compiling vue files, not the actual vue runtime which needs to be added separately. Are you referring to the runtimes or these build packages? For runtimes I prefer option 3 as well for build packages I might lean towards including them as they are quite small anyways.

I was planning to use the zero-config https://github.com/kellymears/laravel-mix-wp-blocks and use it on a `blocks.js` file where you can add, disable, modify blocks and styles. It comes with JSX transpiling but the react runtime is in WP core.

**Comment by oxyc** [comment]
Extracted some generic gutenberg styles to a tailwind plugin https://github.com/generoi/tailwindcss-gutenberg

**Comment by oxyc** [comment]
I'll merge this as is and we can make further adjustments later.

**Pull request merged**